gigascience/gigadb-website
# Improve accessibility of sortable and filterable tables

# Pull request for issue: #1564

This is a pull request for the following functionalities:

* Grid view tables sorting and filtering improved from an a11y perspective

## How to test?

* Navigate to any of the pages with a grid view, e.g. http://gigadb.gigasciencejournal.com:9170/adminDataset/admin
* Verify the presence of carets in the table headers
* trigger sorting by clicking on a header:
  * Verify that caret changes
  * Press tab key and verify that focus remained in clicked header and was not lost
* Follow similar steps for filter input to check that focus is maintained
* Run playwright tests
```sh
cd playwright
npm i
npm test
```

## How have functionalities been implemented?

* Sortable table headers:
  * Added carets
  * Added aria-sort attribute
  * Maintain focus on sorting
  * Added role button to sorting links
* Filter table inputs
  * Add aria-label property so that they are announced to screen readers
  * Maintain focus on filtering
* These updates are done with a js script that is imported in the CustomGridView component so all tables that use that component get it by default

## Any issues with implementation?

CGridView component (the Yii component, not the custom bult one) seems bugged. Filtering, then sorting and then remove filtering term doesn't seem to restore data

## Any changes to automated tests?

Playwright tests config is changed to prevent running in parallel, with this tests run more slowly but the chance that it hangs the system is lower

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
